### PR TITLE
Database Results Store

### PR DESCRIPTION
**Pull request by bigethan**
Works with postgres and mysql and sqlite. Should be faster than the filesystem and allow for future fun with the data.

Required some refactoring of the file.js interface as well as the endpoint rendering to work with callbacks, which was needed anyway as the file system was not async.

### CODE REVIEW

**Comment by bigethan** [comment]
@trulia/webpagetest-charts any feedback?

File: data_store/db.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ * Asked for a test result, retrieve it.
+ */
+
+var debug   = require('debug')('wpt-api:data_store');
+var moment  = require('moment');
+var async   = require('async');
+var camelCaseKeys = require('camelcase-keys');
+var db = require('any-db');
+
+var config  = {conString: 'postgres://bigethan:@localhost/bigethan'}
```
Comment (alexindigo) [inline data_store/db.js]: Is it just and example file?

File: data_store/db.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+         data = {
+           suiteId: suiteId,
+           tests: result.rows.map(function(r){ return r.test; })
+         };
+         callback(data);
+       });
+  },
+
+  getSuiteTest: function getSuiteTest_anon(suiteId, testId, callback) {
```
Comment (alexindigo) [inline data_store/db.js]: `_anon`? :)

File: package.json [hunk 3]
```diff
@@ -6,7 +6,10 @@
     "start": "node ./bin/www"
   },
   "dependencies": {
-    "async": "0.9.0 ",
+    "any-db": "^2.1.0",
```
Comment (alexindigo) [inline package.json]: Do you have carrots here on purpose?

**Comment by alexindigo** [comment]
Nice hat :)

File: package.json (hunk 3 above)
Comment (bigethan) [inline package.json]: nope, was just leaving the defaults from `npm --save`

**Comment by bigethan** [comment]
thanks :-) (for the review too)

**Comment by alexindigo** [comment]
I'm still understanding the thing, so probably would have more questions

**Comment by alexindigo** [comment]
Btw, everybody can't wait to go to lunch with you :)

**Pull request merged**